Modules/Office365DSC/DSCResources/MSFT_EXOAntiPhishPolicy/MSFT_EXOAntiPhishPolicy.psm1

```diff
@@ -0,0 +1,701 @@
+        [Parameter()]
+        [System.Boolean]
+        $EnableSimilarUsersSafetyTips = $false,
+
+        [Parameter()]
+        [System.Boolean]
+        $EnableTargetedDomainsProtection = $false,
+
+        [Parameter()]
+        [System.Boolean]
+        $EnableTargetedUserProtection = $false,
+
+        [Parameter()]
+        [System.Boolean]
+        $EnableUnusualCharactersSafetyTips = $false,
+
+        [Parameter()]
+        [ValidateSet('Present', 'Absent')]
+        [System.String]
+        $Ensure = 'Present',
+
+        [Parameter()]
+        [System.String[]]
+        $ExcludedDomains = @(),
+
+        [Parameter()]
+        [System.String[]]
+        $ExcludedSenders = @(),
+
+        [Parameter(Mandatory = $true)]
+        [System.Management.Automation.PSCredential]
+        $GlobalAdminAccount,
+
+        [Parameter(Mandatory = $true)]
+        [System.String]
+        $Identity,
+
+        [Parameter()]
+        [System.Boolean]
+        $MakeDefault = $false,
+
+        [Parameter()]
+        [ValidateSet('1', '2', '3', '4')]
+        [System.String]
+        $PhishThresholdLevel = '1',
+
+        [Parameter()]
+        [System.String[]]
+        $TargetedDomainActionRecipients = @(),
+
+        [Parameter()]
+        [ValidateSet('BccMessage', 'Delete', 'MoveToJmf', 'NoAction', 'Quarantine', 'Redirect')]
+        [System.String]
+        $TargetedDomainProtectionAction = 'NoAction',
+
+        [Parameter()]
+        [System.String[]]
+        $TargetedDomainsToProtect = @(),
+
+        [Parameter()]
+        [System.String[]]
+        $TargetedUserActionRecipients = @(),
+
+        [Parameter()]
+        [ValidateSet('BccMessage', 'Delete', 'MoveToJmf', 'NoAction', 'Quarantine', 'Redirect')]
+        [System.String]
+        $TargetedUserProtectionAction = 'NoAction',
+
+        [Parameter()]
+        [System.String[]]
+        $TargetedUsersToProtect = @(),
+
+        [Parameter()]
+        [System.Boolean]
+        $TreatSoftPassAsAuthenticated = $true
+    )
+    Write-Verbose -Message "Testing AntiPhishPolicy for $($Identity)"
+    $CurrentValues = Get-TargetResource @PSBoundParameters
+    $AntiPhishPolicyTestParams = $PSBoundParameters
+    $AntiPhishPolicyTestParams.Remove("GlobalAdminAccount") | out-null
+    $TestResult = Test-Office365DSCParameterState -CurrentValues $CurrentValues `
+        -DesiredValues $PSBoundParameters `
+        -ValuesToCheck $AntiPhishPolicyTestParams.Keys
+    if ($TestResult)
+    {
+        Write-Verbose "Closing Remote PowerShell Sessions"
+        $ClosedPSSessions = (Get-PSSession | Remove-PSSession)
+        Write-Verbose "Global ExchangeOnlineSession status: `n"
+        Write-Verbose "$( Get-PSSession -ErrorAction SilentlyContinue | Where-Object Name -eq 'ExchangeOnline' | Out-String)"
+    }
+
+    return $TestResult
+}
+
+function Export-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.String])]
+    param
+    (
+        [Parameter()]
+        [System.String]
```

Comment:
Only Mandatory parameters are expected in the method's signature

Reply:
I'm not  sure I understand what you're saying here.  Are you saying that the Export-TargetResource functions only need to have mandatory parameters in the param block?

Reply:
Removed non-Mandatory parameters from Export-TargetResource functions.